pages/dkp/kommander/1.4/projects/platform-services/platform-services-catalog/kudo-spark/index.md

```diff
@@ -0,0 +1,518 @@
+are a cluster-scoped resource which requires a manual cleanup when all KUDO Spark Operator instances are uninstalled.
+
+To completely remove KUDO Spark Operator from a Kubernetes cluster:
+
+1. Wait for the running jobs to complete or terminate them
+
+   ```
+   kubectl delete sparkapplications --all
+   kubectl delete scheduledsparkapplications --all
+   ```
+
+1. Uninstall each KUDO Spark Operator instance:
+
+   ```
+   kubectl kudo uninstall --instance spark-instance --namespace spark
+   ```
+
+1. Remove Spark Applications CRDs:
+
+   ```
+   kubectl delete crds sparkapplications.sparkoperator.k8s.io scheduledsparkapplications.sparkoperator.k8s.io
+   ```
+
+#### Installing multiple Spark Operator Instances
+
+Optionally, create dedicated namespaces for installing KUDO Spark instances(e.g. `spark-operator-1` and `spark-operator-2` in this example):
+```bash
+kubectl create ns spark-operator-1 && kubectl create ns spark-operator-2
+```
+```
+kubectl kudo install spark --instance=spark-1 --namespace spark-operator-1 -p sparkJobNamespace=spark-operator-1
+kubectl kudo install spark --instance=spark-2 --namespace spark-operator-2 -p sparkJobNamespace=spark-operator-2
+```
+
+The above commands will install two Spark Operators in two different namespaces. Spark Applications submitted to a specific
+namespace will be handled by the Operator installed in the same namespace. This is achieved by explicitly setting
+the `sparkJobNamespace` parameter to corresponding operator namespace.
+
+#### KUDO Spark Operator Upgrade Prior to Konvoy Upgrade or Install
+
+Custom Resource Definitions of KUDO Spark Operator versions prior to 3.0.0-1.1.0 do not specify default values for `x-kubernetes-list-map-keys` properties and will fail validation on Kubernetes versions 1.18.x and later.
+
+Perform these steps prior to upgrading or installing Konvoy to prevent or mitigate disruption of currently-running Spark jobs and invalidating Spark CRDs:
+
+1. Wait for the KUDO Spark Operator jobs to finish, or terminate the running jobs.
+1. [Uninstall the KUDO Spark Operator](#uninstalling-the-spark-operator).
+1. [Install the new KUDO Spark version](#installation).
+1. [Upgrade](../../../../../../konvoy/latest/upgrade) or [install](../../../../../../konvoy/latest/install) Konvoy.
+
+### Submitting Spark Applications
+
+To submit Spark Application you should have KUDO Spark Operator installed.
+
+In order to deploy a Spark Application to Kubernetes using the KUDO Spark Operator, it should be described as a Kubernetes object. To do that, create a specification in `yaml` format with all the necessary configuration required for the application.
+
+Let's take a simple `SparkPi` application as an example. The `yaml` specification could be found here: [spark-pi.yaml](resources/spark-pi.yaml). This example assumes that you installed KUDO spark to the `spark` namespace.
```

Comment:
This link is broken (both here, and in the kudo.dev site) - closest I found was here, I guess? But, I don't know if you want to do this, it looks like the version may be a bit outdated @kaiwalyajoshi 

```suggestion
Let's take a simple `SparkPi` application as an example. The `yaml` specification could be found here: [spark-pi.yaml](https://raw.githubusercontent.com/kudobuilder/operators/master/repository/spark/docs/2.4.5-1.0.1/resources/spark-pi.yaml). This example assumes that you installed KUDO spark to the `spark` namespace.
```

Reply:
replaced with the correct link